src/components/ui/popupwindow.tsx

```diff
@@ -0,0 +1,285 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Modal,
+  View,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  ScrollView,
+  KeyboardAvoidingView,
+  Platform,
+} from 'react-native';
+import { styled } from 'nativewind';
+import { Ionicons } from '@expo/vector-icons';
+import DateTimePickerModal from 'react-native-modal-datetime-picker';
+import Animated, {
+  useSharedValue,
+  withTiming,
+  useAnimatedStyle,
+  Easing,
+} from 'react-native-reanimated';
+
+interface TextFieldConfig {
+  placeholder: string;
+}
+
+interface Group {
+  name: string;
+  color: string;
+}
+
+interface AddTaskModalUIProps {
+  visible: boolean;
+  onClose: () => void;
+  onSave: () => void;
+  showDatePicker?: boolean;
+  showGroupSelector?: boolean;
```

Comment:
You should implement DatePicker and Group selector as separate components, and get them as props of React.ReactNode type